Refactor/test runner

PascalPrecht opened the pull request.

jrainville commented on packages/embark-utils/src/index.js:
```diff
@@ -7,6 +7,26 @@ const {canonicalHost, defaultCorsHost, defaultHost, dockerHostSwap, isDocker} =
 const toposortGraph = require('./toposort');
+import { unitRegex } from './constants';
+import * as AddressUtils from './addressUtils';
+const web3 = require("web3");
```
I wouldn't require web3 at the top of the file. `web3` is the reason we started putting requires inside the function, because some processes needed to use an util function like `checkIsAvailable`, but it ended up loading `web3` (it's not cached in that case because it's another process), so it slowed down the whole process.

jrainville commented on packages/embark-utils/src/web3Utils.ts:
```diff
@@ -0,0 +1,51 @@
+const Web3EthAbi = require("web3-eth-abi");
+const Web3 = require("web3");
+
+export function getHexBalanceFromString(balanceString: string, web3: any) {
```
Since we require Web3 in this file, and that's the same thing we do in `utils.js`, shouldn't we just remove the need to pass web3 as a parameter?
I think we decided to pass it as a param so as to not require web3, but now we don't care since we already do.

jrainville commented on packages/embark-utils/src/web3Utils.ts:
```diff
@@ -0,0 +1,51 @@
+import { __ } from "i18n";
+import { balanceRegex } from "./constants";
+const Web3EthAbi = require("web3-eth-abi");
+const Web3 = require("web3");
```
I would make it so that this file is the only utils file that requires Web3, and when you want to use those functions, you do `import Web3Utils from 'embark-utils/web3Utils'`, so that way you only require web3 when you need it. Does it make sense? Also, is it doable?

PascalPrecht commented:
Thanks for the great feedback @jrainville 

I've updated the PR so that web3 is only imported in web3Utils, and moved web3 dependent functions into web3Utils as well.

I'd like to put the `embark-utils/SomethingUtils` APIs in a separate PR as this most likely requires us to touch a lot of files in the system again.

PascalPrecht commented:
CI is failing due to typecheck errors unrelated to my changes
